qiskit/transpiler/__init__.py

```diff
@@ -19,8 +19,364 @@
+   All said, unrolling to the basis set of gates leads to an increase in the depth of a
+   quantum circuit and the number of gates.
+
+   It is important to highlight two special cases:
+
+   1. A SWAP gate is not a native gate on the IBM Q devices, and must be decomposed into
+      three CNOT gates:
+
+      .. jupyter-execute::
+
+         swap_circ = QuantumCircuit(2)
+         swap_circ.swap(0, 1)
+         swap_circ.decompose().draw(output='mpl')
+
+      As a product of three CNOT gates, SWAP gates are expensive operations to perform on a
+      noisy quantum devices.  However, such operations are usually necessary for embedding a
+      circuit into the limited entangling gate connectivities of actual devices.  Thus,
+      minimizing the number of SWAP gates in a circuit is a primary goal in the
+      transpilation process.
+
+
+   2. A Toffoli, or controlled-controlled-not gate (`ccx`), is a three-qubit gate.  Given
+      that our basis gate set includes only single- and two-qubit gates, it is obvious that
+      this gate must be decomposed.  This decomposition is quite costly:
+
+      .. jupyter-execute::
+
+         ccx_circ = QuantumCircuit(3)
+         ccx_circ.ccx(0, 1, 2)
+         ccx_circ.decompose().draw(output='mpl')
+
+      For every Toffoli gate in a quantum circuit, the IBM Quantum hardware may execute up to
+      six CNOT gates, and a handful of single-qubit gates.  From this example, it should be
+      clear that any algorithm that makes use of multiple Toffoli gates will end up as a
+      circuit with large depth and will therefore be appreciably affected by noise and gate
+      errors.
+
+
+   .. raw:: html
+
+      <br>
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Initial Layout**
+
+   Quantum circuits are abstract entities whose qubits are "virtual" representations of actual
+   qubits used in computations.  We need to be able to map these virtual qubits in a one-to-one
+   manner to the "physical" qubits in an actual quantum device.
+
+   .. image:: /source_images/mapping.png
+
+   .. raw:: html
+
+      <br><br>
+
+   By default, qiskit will do this mapping for you.  The choice of mapping depends on the
+   properties of the circuit, the particular device you are targeting, and the optimization
+   level that is chosen.  The basic mapping strategies are the following:
+
+   - **Trivial layout**: Map virtual qubits to the same numbered physical qubit on the device,
+     i.e. `[0,1,2,3,4]` -> `[0,1,2,3,4]` (default in `optimization_level=0`).
+
+   - **Dense layout**: Find the sub-graph of the device with same number of qubits as the circuit
+     with the greatest connectivity (default in `optimization_level=1`).
+
+   - **Noise adaptive layout**: Uses the noise properties of the device, in concert with the
+     circuit properties, to generate the layout with the best noise properties (default in
+     `optimization_level=2` and `optimization_level=3`).
```

Comment:
This isn't quite true anymore. Dense layout is the default in level 2 and 3 now that it's noise aware:

https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/preset_passmanagers/level2.py#L80
https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/preset_passmanagers/level3.py#L84

Reply:
Ahh another change.  Ok.